src/specializations/Ray.jl

```diff
@@ -11,85 +11,24 @@
-    t₁(x) = (1 // 2) * x + (1 // 2)
-    t₁′(x) = (1 // 2)
-    t₂(x) = x / (1 - x^2)
-    t₂′(x) = (1 + x^2) / (1 - x^2)^2
-    t = t₂ ∘ t₁
-    t′(x) = t₂′(t₁(x)) * t₁′(x)
-
-    # Integrate f along the Ray
-    domainunits = _units(ray(0))
-    integrand(x) = f(ray(t(x))) * t′(x) * domainunits
-    return sum(w .* integrand(x) for (w, x) in zip(ws, xs))
-end
-
-function integral(
-        f,
-        ray::Meshes.Ray,
-        rule::GaussKronrod;
-        diff_method::DM = Analytical(),
-        FP::Type{T} = Float64
-) where {DM <: DifferentiationMethod, T <: AbstractFloat}
-    _guarantee_analytical(Meshes.Ray, diff_method)
-
-    # Normalize the Ray s.t. ray(t) is distance t from origin point
-    ray = Meshes.Ray(ray.p, Meshes.unormalize(ray.v))
-
-    # Integrate f along the Ray
-    domainunits = _units(ray(0))
-    integrand(t) = f(ray(t)) * domainunits
-    return QuadGK.quadgk(integrand, zero(FP), FP(Inf); rule.kwargs...)[1]
-end
-
-function integral(
-        f,
-        ray::Meshes.Ray,
-        rule::HAdaptiveCubature;
-        diff_method::DM = Analytical(),
-        FP::Type{T} = Float64
-) where {DM <: DifferentiationMethod, T <: AbstractFloat}
-    _guarantee_analytical(Meshes.Ray, diff_method)
-
-    # Normalize the Ray s.t. ray(t) is distance t from origin point
-    ray = Meshes.Ray(ray.p, Meshes.unormalize(ray.v))
-
-    # Domain transformation: x ∈ [0,1] ↦ t ∈ [0,∞)
-    t(x) = x / (1 - x^2)
-    t′(x) = (1 + x^2) / (1 - x^2)^2
-
-    # Integrate f along the Ray
-    domainunits = _units(ray(0))
-    integrand(xs) = f(ray(t(xs[1]))) * t′(xs[1]) * domainunits
-
-    # HCubature doesn't support functions that output Unitful Quantity types
-    # Establish the units that are output by f
-    testpoint_parametriccoord = _zeros(FP, 1)
-    integrandunits = Unitful.unit.(integrand(testpoint_parametriccoord))
-    # Create a wrapper that returns only the value component in those units
-    uintegrand(uv) = Unitful.ustrip.(integrandunits, integrand(uv))
-    # Integrate only the unitless values
-    value = HCubature.hcubature(uintegrand, _zeros(FP, 1), _ones(FP, 1); rule.kwargs...)[1]
-
-    # Reapply units
-    return value .* integrandunits
+        rule::IntegrationRule;
+        kwargs...
+)
+    # Generate a _ParametricGeometry whose parametric function spans the domain [0, 1]
+    param_ray = _ParametricGeometry(_parametric(ray), Meshes.paramdim(ray))
+
+    # Integrate the _ParametricGeometry using the standard methods
+    return _integral(f, param_ray, rule; kwargs...)
 end
 
 ################################################################################
-#                               jacobian
+#                                 Parametric
 ################################################################################
 
-_has_analytical(::Type{T}) where {T <: Meshes.Ray} = true
+# Map argument domain from [0, 1] to [0, ∞) for (::Ray)(t)
+# f(t) = t / 1 - t^2)
+# f'(t) = (t^2 + 1) / (1 - t^2)^2
```

Comment:
```suggestion
# f(t) = t / (1 - t^2)
```

Reply:
I left `f'(t)` as a hint to future-me in case I wanted to try implementing an analytical derivative. That's no longer the case, so I went ahead and removed both lines.